common/common.go

```diff
@@ -66,3 +69,33 @@ type Parameters map[string]string
 func (s Parameters) Equals(is Parameters) bool {
 	return reflect.DeepEqual(s, is)
 }
+
+// Write file to temp and atomically move when everything else succeeds.
+// This function is taken from
+//   https://github.com/youtube/vitess/blob/master/go/ioutil2/ioutil.go
+// Copyright 2012, Google Inc. BSD-license
```

Comment:
It would be slightly better to drop a copy of the BSD license somewhere into the repo if you're OK with that (how about a `licenses` directory) and reference this from here, as conceptually you are supposed to include the full text of the license somewhere (because it carries the disclaimer).